server/mdm/apple/cert.go

```diff
@@ -0,0 +1,123 @@
+package apple_mdm
+
+import (
+	"bytes"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+
+	"github.com/micromdm/scep/v2/depot"
+)
+
+const defaultFleetDMAPIURL = "https://fleetdm.com"
+
+const getSignedAPNSCSRPath = "/api/v1/get_signed_apns_csr"
+
+// EmailAddressOID defined by https://oidref.com/1.2.840.113549.1.9.1
+var EmailAddressOID = []int{1, 2, 840, 113549, 1, 9, 1}
+
+// GenerateAPNSCSRKey generates a APNS csr to be sent to fleetdm.com and returns a csr and key.
+func GenerateAPNSCSRKey(email, org string) (*x509.CertificateRequest, *rsa.PrivateKey, error) {
```

Comment:
a question for my own understanding: the spec mentions:

> The email, org, and metadata about the APNs certificate should be saved in the fleetdm.com database for future references, analysis, and auditing.

Do you know how this is meant to work? does that part apply only when the website is calling this API?

Reply:
I don't know how this is expected to work. afaik, the fleetdm.com api endpoint has not been implemented yet, though there is a draft pr. See https://github.com/fleetdm/fleet/issues/8223.

Reply:
understood, thanks! I will ask around.